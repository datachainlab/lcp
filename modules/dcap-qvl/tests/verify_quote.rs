// use dcap_qvl::{quote::Quote, verify::verify, QuoteCollateralV3};
// use scale::Decode;

// #[test]
// fn could_parse_sgx_quote() {
//     let raw_quote = include_bytes!("../sample/sgx_quote").to_vec();
//     let raw_quote_collateral = include_bytes!("../sample/sgx_quote_collateral").to_vec();
//     let now = 1699301000u64;

//     let quote = Quote::decode(&mut &raw_quote[..]).unwrap();
//     insta::assert_debug_snapshot!(quote);

//     let quote_collateral =
//         QuoteCollateralV3::decode(&mut raw_quote_collateral.as_slice()).expect("decodable");
//     let tcb_status = verify(&raw_quote, &quote_collateral, now).expect("verify");

//     assert_eq!(tcb_status.status, "ConfigurationAndSWHardeningNeeded");
//     assert_eq!(
//         tcb_status.advisory_ids,
//         ["INTEL-SA-00289", "INTEL-SA-00615"]
//     );
// }

// #[test]
// fn could_parse_tdx_quote() {
//     let raw_quote = include_bytes!("../sample/tdx_quote");
//     let raw_quote_collateral = include_bytes!("../sample/tdx_quote_collateral");
//     let now = 1725258675u64;

//     let quote = Quote::decode(&mut &raw_quote[..]).unwrap();
//     insta::assert_debug_snapshot!(quote);

//     let quote_collateral = QuoteCollateralV3::decode(&mut raw_quote_collateral.as_slice()).unwrap();
//     let tcb_status = verify(raw_quote, &quote_collateral, now).unwrap();
//     assert_eq!(tcb_status.status, "UpToDate");
//     assert!(tcb_status.advisory_ids.is_empty());
// }
