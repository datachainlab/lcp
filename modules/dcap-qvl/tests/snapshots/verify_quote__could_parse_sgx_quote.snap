---
source: tests/verify_quote.rs
assertion_line: 11
expression: quote
---
Quote {
    header: Header {
        version: 3,
        attestation_key_type: 2,
        tee_type: 0,
        qe_svn: 10,
        pce_svn: 15,
        qe_vendor_id: [
            147,
            154,
            114,
            51,
            247,
            156,
            76,
            169,
            148,
            10,
            13,
            179,
            149,
            127,
            6,
            7,
        ],
        user_data: [
            57,
            135,
            98,
            46,
            230,
            150,
            138,
            84,
            151,
            124,
            134,
            38,
            239,
            71,
            18,
            53,
            0,
            0,
            0,
            0,
        ],
    },
    report: SgxEnclave(
        EnclaveReport {
            cpu_svn: [
                11,
                11,
                26,
                24,
                255,
                255,
                4,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
            ],
            misc_select: 0,
            reserved1: [
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
            ],
            attributes: [
                5,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                231,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
            ],
            mr_enclave: [
                51,
                216,
                115,
                109,
                183,
                86,
                237,
                73,
                151,
                224,
                75,
                163,
                88,
                210,
                120,
                51,
                24,
                143,
                25,
                50,
                255,
                123,
                29,
                21,
                105,
                4,
                211,
                245,
                96,
                69,
                47,
                187,
            ],
            reserved2: [
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
            ],
            mr_signer: [
                129,
                95,
                66,
                241,
                28,
                246,
                68,
                48,
                195,
                11,
                171,
                120,
                22,
                186,
                89,
                106,
                29,
                160,
                19,
                12,
                59,
                2,
                139,
                103,
                49,
                51,
                166,
                108,
                249,
                163,
                224,
                230,
            ],
            reserved3: [
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
            ],
            isv_prod_id: 0,
            isv_svn: 0,
            reserved4: [
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
            ],
            report_data: [
                72,
                101,
                108,
                108,
                111,
                44,
                32,
                119,
                111,
                114,
                108,
                100,
                33,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
            ],
        },
    ),
    auth_data: V3(
        AuthDataV3 {
            ecdsa_signature: [
                109,
                221,
                149,
                2,
                163,
                9,
                61,
                34,
                191,
                41,
                207,
                6,
                98,
                214,
                233,
                82,
                252,
                126,
                159,
                64,
                72,
                44,
                208,
                222,
                108,
                33,
                129,
                105,
                175,
                247,
                246,
                137,
                41,
                77,
                5,
                24,
                237,
                66,
                133,
                101,
                54,
                133,
                233,
                250,
                254,
                64,
                100,
                59,
                69,
                137,
                178,
                25,
                7,
                182,
                76,
                252,
                148,
                39,
                186,
                84,
                35,
                145,
                45,
                119,
            ],
            ecdsa_attestation_key: [
                220,
                226,
                185,
                31,
                236,
                210,
                250,
                37,
                84,
                109,
                65,
                193,
                213,
                12,
                109,
                33,
                226,
                138,
                224,
                68,
                33,
                83,
                208,
                146,
                165,
                5,
                253,
                75,
                2,
                185,
                189,
                57,
                82,
                230,
                233,
                12,
                36,
                5,
                211,
                227,
                73,
                238,
                241,
                253,
                88,
                80,
                132,
                14,
                43,
                232,
                59,
                196,
                254,
                101,
                145,
                113,
                214,
                21,
                8,
                95,
                114,
                213,
                123,
                127,
            ],
            qe_report: [
                11,
                11,
                26,
                24,
                255,
                255,
                4,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                21,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                231,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                150,
                179,
                71,
                166,
                78,
                90,
                4,
                94,
                39,
                54,
                156,
                38,
                230,
                220,
                218,
                81,
                253,
                124,
                133,
                14,
                155,
                58,
                58,
                121,
                231,
                24,
                244,
                50,
                97,
                222,
                225,
                228,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                140,
                79,
                87,
                117,
                215,
                150,
                80,
                62,
                150,
                19,
                127,
                119,
                198,
                138,
                130,
                154,
                0,
                86,
                172,
                141,
                237,
                112,
                20,
                11,
                8,
                27,
                9,
                68,
                144,
                197,
                123,
                255,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                1,
                0,
                10,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                194,
                97,
                187,
                136,
                46,
                84,
                42,
                168,
                215,
                249,
                233,
                154,
                0,
                239,
                203,
                17,
                207,
                46,
                230,
                111,
                169,
                198,
                134,
                31,
                146,
                48,
                211,
                248,
                3,
                162,
                117,
                253,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
            ],
            qe_report_signature: [
                191,
                176,
                167,
                89,
                204,
                134,
                78,
                136,
                25,
                241,
                183,
                210,
                106,
                189,
                231,
                118,
                49,
                129,
                110,
                36,
                205,
                192,
                47,
                36,
                170,
                152,
                111,
                212,
                7,
                204,
                131,
                152,
                69,
                206,
                21,
                186,
                124,
                42,
                235,
                14,
                109,
                22,
                136,
                218,
                25,
                245,
                163,
                146,
                197,
                12,
                5,
                175,
                140,
                108,
                79,
                98,
                45,
                8,
                234,
                22,
                183,
                183,
                43,
                71,
            ],
            qe_auth_data: Data {
                data: [
                    0,
                    1,
                    2,
                    3,
                    4,
                    5,
                    6,
                    7,
                    8,
                    9,
                    10,
                    11,
                    12,
                    13,
                    14,
                    15,
                    16,
                    17,
                    18,
                    19,
                    20,
                    21,
                    22,
                    23,
                    24,
                    25,
                    26,
                    27,
                    28,
                    29,
                    30,
                    31,
                ],
                _marker: PhantomData<u16>,
            },
            certification_data: CertificationData {
                cert_type: 5,
                body: "-----BEGIN CERTIFICATE-----\nMIIEjTCCBDSgAwIBAgIVAIG3dzK3YemOubljpKvR5bm/XdjWMAoGCCqGSM49BAMC\nMHExIzAhBgNVBAMMGkludGVsIFNHWCBQQ0sgUHJvY2Vzc29yIENBMRowGAYDVQQK\nDBFJbnRlbCBDb3Jwb3JhdGlvbjEUMBIGA1UEBwwLU2FudGEgQ2xhcmExCzAJBgNV\nBAgMAkNBMQswCQYDVQQGEwJVUzAeFw0yMzA5MjAyMTUzNDNaFw0zMDA5MjAyMTUz\nNDNaMHAxIjAgBgNVBAMMGUludGVsIFNHWCBQQ0sgQ2VydGlmaWNhdGUxGjAYBgNV\nBAoMEUludGVsIENvcnBvcmF0aW9uMRQwEgYDVQQHDAtTYW50YSBDbGFyYTELMAkG\nA1UECAwCQ0ExCzAJBgNVBAYTAlVTMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE\nkgmE7N3D+RspyaCZ2YoDTLDCuh5pnvAu4crPn2uAGujq9tOgwU8/y7jttShCB603\nU6r+h9ayOk2nZ9jewk25lqOCAqgwggKkMB8GA1UdIwQYMBaAFNDoqtp11/kuSReY\nPHsUZdDV8llNMGwGA1UdHwRlMGMwYaBfoF2GW2h0dHBzOi8vYXBpLnRydXN0ZWRz\nZXJ2aWNlcy5pbnRlbC5jb20vc2d4L2NlcnRpZmljYXRpb24vdjQvcGNrY3JsP2Nh\nPXByb2Nlc3NvciZlbmNvZGluZz1kZXIwHQYDVR0OBBYEFIW4KX263PRxYJah2Cfj\nAlrcvAC9MA4GA1UdDwEB/wQEAwIGwDAMBgNVHRMBAf8EAjAAMIIB1AYJKoZIhvhN\nAQ0BBIIBxTCCAcEwHgYKKoZIhvhNAQ0BAQQQ0E7AbU5tktyQ0K089e4t3zCCAWQG\nCiqGSIb4TQENAQIwggFUMBAGCyqGSIb4TQENAQIBAgELMBAGCyqGSIb4TQENAQIC\nAgELMBAGCyqGSIb4TQENAQIDAgECMBAGCyqGSIb4TQENAQIEAgECMBEGCyqGSIb4\nTQENAQIFAgIA/zAQBgsqhkiG+E0BDQECBgIBATAQBgsqhkiG+E0BDQECBwIBADAQ\nBgsqhkiG+E0BDQECCAIBADAQBgsqhkiG+E0BDQECCQIBADAQBgsqhkiG+E0BDQEC\nCgIBADAQBgsqhkiG+E0BDQECCwIBADAQBgsqhkiG+E0BDQECDAIBADAQBgsqhkiG\n+E0BDQECDQIBADAQBgsqhkiG+E0BDQECDgIBADAQBgsqhkiG+E0BDQECDwIBADAQ\nBgsqhkiG+E0BDQECEAIBADAQBgsqhkiG+E0BDQECEQIBDTAfBgsqhkiG+E0BDQEC\nEgQQCwsCAv8BAAAAAAAAAAAAADAQBgoqhkiG+E0BDQEDBAIAADAUBgoqhkiG+E0B\nDQEEBAYAoGcRAAAwDwYKKoZIhvhNAQ0BBQoBADAKBggqhkjOPQQDAgNHADBEAiBm\nSMZEtlQEjnZgGa192W3ArnZ3iyY6ckM/sTsXxCRmJgIgLf20tZHNw3a1b31JDSOW\nE6wesxoAmTeqJGRqZl621qI=\n-----END CERTIFICATE-----\n-----BEGIN CERTIFICATE-----\nMIICmDCCAj6gAwIBAgIVANDoqtp11/kuSReYPHsUZdDV8llNMAoGCCqGSM49BAMC\nMGgxGjAYBgNVBAMMEUludGVsIFNHWCBSb290IENBMRowGAYDVQQKDBFJbnRlbCBD\nb3Jwb3JhdGlvbjEUMBIGA1UEBwwLU2FudGEgQ2xhcmExCzAJBgNVBAgMAkNBMQsw\nCQYDVQQGEwJVUzAeFw0xODA1MjExMDUwMTBaFw0zMzA1MjExMDUwMTBaMHExIzAh\nBgNVBAMMGkludGVsIFNHWCBQQ0sgUHJvY2Vzc29yIENBMRowGAYDVQQKDBFJbnRl\nbCBDb3Jwb3JhdGlvbjEUMBIGA1UEBwwLU2FudGEgQ2xhcmExCzAJBgNVBAgMAkNB\nMQswCQYDVQQGEwJVUzBZMBMGByqGSM49AgEGCCqGSM49AwEHA0IABL9q+NMp2IOg\ntdl1bk/uWZ5+TGQm8aCi8z78fs+fKCQ3d+uDzXnVTAT2ZhDCifyIuJwvN3wNBp9i\nHBSSMJMJrBOjgbswgbgwHwYDVR0jBBgwFoAUImUM1lqdNInzg7SVUr9QGzknBqww\nUgYDVR0fBEswSTBHoEWgQ4ZBaHR0cHM6Ly9jZXJ0aWZpY2F0ZXMudHJ1c3RlZHNl\ncnZpY2VzLmludGVsLmNvbS9JbnRlbFNHWFJvb3RDQS5kZXIwHQYDVR0OBBYEFNDo\nqtp11/kuSReYPHsUZdDV8llNMA4GA1UdDwEB/wQEAwIBBjASBgNVHRMBAf8ECDAG\nAQH/AgEAMAoGCCqGSM49BAMCA0gAMEUCIQCJgTbtVqOyZ1m3jqiAXM6QYa6r5sWS\n4y/G7y8uIJGxdwIgRqPvBSKzzQagBLQq5s5A70pdoiaRJ8z/0uDz4NgV91k=\n-----END CERTIFICATE-----\n-----BEGIN CERTIFICATE-----\nMIICjzCCAjSgAwIBAgIUImUM1lqdNInzg7SVUr9QGzknBqwwCgYIKoZIzj0EAwIw\naDEaMBgGA1UEAwwRSW50ZWwgU0dYIFJvb3QgQ0ExGjAYBgNVBAoMEUludGVsIENv\ncnBvcmF0aW9uMRQwEgYDVQQHDAtTYW50YSBDbGFyYTELMAkGA1UECAwCQ0ExCzAJ\nBgNVBAYTAlVTMB4XDTE4MDUyMTEwNDUxMFoXDTQ5MTIzMTIzNTk1OVowaDEaMBgG\nA1UEAwwRSW50ZWwgU0dYIFJvb3QgQ0ExGjAYBgNVBAoMEUludGVsIENvcnBvcmF0\naW9uMRQwEgYDVQQHDAtTYW50YSBDbGFyYTELMAkGA1UECAwCQ0ExCzAJBgNVBAYT\nAlVTMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEC6nEwMDIYZOj/iPWsCzaEKi7\n1OiOSLRFhWGjbnBVJfVnkY4u3IjkDYYL0MxO4mqsyYjlBalTVYxFP2sJBK5zlKOB\nuzCBuDAfBgNVHSMEGDAWgBQiZQzWWp00ifODtJVSv1AbOScGrDBSBgNVHR8ESzBJ\nMEegRaBDhkFodHRwczovL2NlcnRpZmljYXRlcy50cnVzdGVkc2VydmljZXMuaW50\nZWwuY29tL0ludGVsU0dYUm9vdENBLmRlcjAdBgNVHQ4EFgQUImUM1lqdNInzg7SV\nUr9QGzknBqwwDgYDVR0PAQH/BAQDAgEGMBIGA1UdEwEB/wQIMAYBAf8CAQEwCgYI\nKoZIzj0EAwIDSQAwRgIhAOW/5QkR+S9CiSDcNoowLuPRLsWGf/Yi7GSX94BgwTwg\nAiEA4J0lrHoMs+Xo5o/sX6O9QWxHRAvZUGOdRQ7cvqRXaqI=\n-----END CERTIFICATE-----\n\0",
            },
        },
    ),
}
